### PR TITLE
Changing autofetch to a string config which has "current", "all" and

### LINKED ISSUE #110811: Git: Allow option to "fetch --all" on autofetch
VScode has a "git.autofetch" option which periodically fetches from upstream. However, when forking a project (which a lot of us do) you will only ever fetch from `origin`. In my case I want to periodically fetch from upstream too so I can have up-to-date changes from others working on the same project.

On the command line the change would be adding `--all`.  In terms of the built-in extension the change would be here:
https://github.com/microsoft/vscode/blob/master/extensions/git/src/autofetch.ts#L102

Today `fetchDefault` is always called. [`fetchAll`](https://github.com/microsoft/vscode/blob/master/extensions/git/src/repository.ts#L1310-L1312) exists and could be called instead if an option is set to true.

## Example Config
It would make sense for autoFetch to take a series of options like, `"all", "current", "off"`. But im assuming we can't modify this option now as it's being used.

```json
  "git.autofetch": true, // Most likely can't change this now
  "git.autofetchAll": true, // fetchAll when fetching (overriding above)
```

### PR DESCRIPTION
**Pull request by jasonwilliams**
Changing autofetch to a string config which has "current", "all" and "off".
This allows users to fetch from all remotes instead of just their origin.
For developers working on a Fork, this should make extensions work more intuitively.

This PR fixes #110811

### CODE REVIEW

**Comment by jasonwilliams** [comment]
@eamodio is this the sort of change you was expecting? 
I’m happy to make changes

**Comment by eamodio** [comment]
@jasonwilliams Thanks for taking a shot at this! Although, I meant to change `git.autofetch` to be a `boolean | 'all'`, so in the json it would be like:

```json
"git.autofetch": {
  "anyOf": [
    {
      "type": "boolean"
    },
    {
      "type": "string",
        "enum": [
          "all"
        ]
    }
  ],
  "scope": "resource",
  "description": "%config.autofetch%",
  "default": false,
  "tags": [
    "usesOnlineServices"
  ]
},
```

So that way any existing value will continue to work and we can additionally set the `all` value.

**Comment by jasonwilliams** [comment]
@eamodio I've made the relevant changes now

File: extensions/git/src/autofetch.ts [hunk 1]
```diff
@@ -70,10 +71,18 @@ export class AutoFetcher {
 	private onConfiguration(): void {
 		const gitConfig = workspace.getConfiguration('git', Uri.file(this.repository.root));
 
-		if (gitConfig.get<boolean>('autofetch') === false) {
-			this.disable();
-		} else {
-			this.enable();
+		switch (gitConfig.get<string | boolean>('autofetch')) {
```
Comment (eamodio) [inline extensions/git/src/autofetch.ts:74]: Don't need to change this -- but FYI, you can change this to:
```ts
gitConfig.get<'all' | boolean>('autofetch')
```

to be more specific.

**Comment by eamodio** [comment]
@jasonwilliams Thanks so much for the contribution! 👍

**Pull request merged**